storage/innobase/trx/trx0purge.cc

```diff
@@ -437,6 +438,9 @@ trx_purge_truncate_rseg_history(
 	mtr_t		mtr;
 	trx_id_t	undo_trx_no;
 
+	if (log_is_in_distress())
+		return;
```

Comment:
I was wondering if an early return from this function could cause some corruption something. It looks like the answer is no: in `trx_undo_free_page()` we would decrement `rseg.curr_size` for each page that is being freed, and `trx_purge_truncate_history()` is referring to that count in undo tablespace truncation.

This would be good to mention in a source code comment.

Reply:
Done.